### PR TITLE
Refactor .drone.yml

### PR DESCRIPTION
**Pull request by assafmo**
Matching PRs:
https://github.com/enigmampc/enigma-p2p/pull/289
https://github.com/enigmampc/enigma-core/pull/279

- Until we have a good enough testing framework: Test that a new worker joins the network and syncs correctly. (Cc: @lenak25)
- Add a cleanup step to the CI.
- Small refactoring.

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -71,9 +72,12 @@ steps:
       - make build-contract DOCKER_TAG=$DOCKER_TAG
       - make build-client DOCKER_TAG=$DOCKER_TAG
       - export NODES=3
-      - SGX_MODE=HW docker-compose -f docker-compose.yml -f docker-compose.hw.yml up --scale worker=$NODES --exit-code-from client & sleep 240 && docker-compose exec -T client make test && export RESULT=$? || export RESULT=$?
-      - docker-compose -f docker-compose.yml -f docker-compose.hw.yml down -v --rmi all || true
-      - if [ $RESULT -ne 0 ]; then exit 1; fi
+      - SGX_MODE=HW docker-compose -f docker-compose.yml -f docker-compose.hw.yml up --scale worker=$NODES &
+      - sleep 240
+      - docker-compose exec -T client make test
```
Comment (lacabra) [inline .drone.yml:77]: @assafmo: if any of the regular tests fail here, nothing is going to catch that error, so the CI will report success when the tests will fail. The `docker-compose exec` does not return the error code from the command it executes.

**Review by lacabra — CHANGES_REQUESTED** [review]

File: .drone.yml (hunk 1 above)
Comment (assafmo) [reply .drone.yml:77]: @lacabra Are you sure?
```bash
➜  docker-environment git:(develop) docker-compose exec -T client bash -c 'exit 7' ; echo $?
7
```

Comment (lacabra) [reply .drone.yml:77]: Ok, I'm convinced. I could have sworn that wasn't the case before, but it clearly works now. Thank you

**Review by lacabra — CHANGES_REQUESTED** [review]
For some obscure reason, the removal of images is not working now, see the [logs](https://drone.enigma.co/enigmampc/enigma-contract/289/1/3) which clutters the server significantly:

```
18 | Removing image enigmampc/key_management_hw:
19 | Failed to remove image for service km: 400 Client Error: Bad Request ("invalid reference format")
20 | Removing image enigmampc/contract:
21 | Failed to remove image for service contract: 400 Client Error: Bad Request ("invalid reference format")
22 | Removing image enigmampc/worker_hw:
23 | Failed to remove image for service worker: 400 Client Error: Bad Request ("invalid reference format")
24 | Removing image enigmampc/worker_hw:
25 | Failed to remove image for service bootstrap: 400 Client Error: Bad Request ("invalid reference format")
26 | Removing image enigmampc/client:
27 | Failed to remove image for service client: 400 Client Error: Bad Request ("invalid reference format")
```

**Comment by lacabra** [comment]
I fixed the above error in da51829 (and added another image removal of an intermediate docker image that is not explictly listed in docker-compose: `enigmampc/core-artifact-hw`). After that step succeeds in the build, I will propagate this change to the other related PRs.

**Review by lacabra — APPROVED** [review]

**Pull request merged**